src/Tools/dotnet-monitor/CollectionRules/Configuration/CustomShortcutsConfigureNamedOptions.cs

```diff
@@ -0,0 +1,225 @@
+
+            return true;
+        }
+
+        private bool InsertCustomTriggerIntoTrigger(CollectionRuleOptions ruleOptions, string name)
+        {
+            IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Trigger)));
+            if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+            {
+                if (!ValidateCustomShortcutExists(_shortcutOptions.CurrentValue.Triggers, section.Value))
+                {
+                    return false;
+                }
+
+                CollectionRuleTriggerOptions options = _shortcutOptions.CurrentValue.Triggers[section.Value];
+                ruleOptions.Trigger = options;
+            }
+
+            return true;
+        }
+
+        private bool InsertCustomFiltersIntoFilterList(CollectionRuleOptions ruleOptions, string name)
+        {
+            bool boundAllFilters = false;
+            int index = 0;
+
+            while (!boundAllFilters)
+            {                
+                IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Filters), index.ToString(CultureInfo.InvariantCulture)));
+                if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+                {
+                    if (!InsertCustomShortcut(_shortcutOptions.CurrentValue.Filters, section.Value, ruleOptions.Filters, index))
+                    {
+                        return false;
+                    }
+                }
+                else if (!section.Exists())
+                {
+                    boundAllFilters = true;
+                }
+
+                ++index;
+            }
+
+            return true;
+        }
+
+        private bool InsertCustomLimitIntoLimit(CollectionRuleOptions ruleOptions, string name)
+        {
+            IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Limits)));
+            if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+            {
+                if (!ValidateCustomShortcutExists(_shortcutOptions.CurrentValue.Limits, section.Value))
+                {
+                    return false;
+                }
+
+                CollectionRuleLimitsOptions options = _shortcutOptions.CurrentValue.Limits[section.Value];
+                ruleOptions.Limits = options;
+            }
+
+            return true;
+        }
+
+        private bool InsertCustomShortcut<T>(IDictionary<string, T> shortcutsOptions, string shortcutKey, List<T> options, int index) where T : class
+        {
+            if (!ValidateCustomShortcutExists(shortcutsOptions, shortcutKey))
+            {
+                return false;
+            }
+
+            options.Insert(index, shortcutsOptions[shortcutKey]);
```

Comment:
So does this mean that the bound action settings are adjacent to the original settings, rather than replacements? Would it be easier if we just replaced the originals?

Reply:
IIRC, the `options` in this case only includes the standard actions. So, if I had four actions (normal, custom, normal, custom), the `options` would have a length of 2. I can't just stick the custom actions on the end, or else we'd have ordering problems, so it would do: (normal, normal) -> (normal, custom, normal) ... and ... (normal, custom, normal) -> (normal, custom, normal, custom). I could be mis-remembering though

Reply:
Would it be reasonable to change CollectionRuleOptions.Actions to include all the actions and then substitute accordingly? We can validate that if a user specifies an action with no shortcut, it fails.

Reply:
This wasn't an intentional decision (more-so just something I discovered organically, and then worked around) - I'm assuming that since `Actions` expects `List<CollectionRuleActionOptions`, and not a list that includes strings, that we can't get the custom shortcuts as part of that list...unless we could force it to just create empty actions as placeholders.